unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,238 @@
+                aggregator.delete_aggregator()
+
+    @staticmethod
+    @pytest.mark.parametrize("test_is_blocking, expected_ret_val",
+                             [(True, TEST_TRANSACTION_ID), (False, None)])
+    def test_delete_aggregator_returns_expected(aggregator, test_is_blocking,
+                                                expected_ret_val):
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+        assert aggregator.delete_aggregator(is_blocking=test_is_blocking) is expected_ret_val
+
+
+@pytest.mark.usefixtures("mock_client_command_buffer_attributes")
+class TestRedisAggregatorExecuteBatchCommands:
+    """Test cases for RedisAggregator's execute_batch_commands method."""
+
+    @staticmethod
+    def test_execute_batch_commands_commands_buffer_length_zero_return_none(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer.buffer_length",
+                   new_callable=PropertyMock, return_value=0):
+            assert aggregator.execute_batch_commands() is None
+
+    @staticmethod
+    def test_execute_batch_commands_not_all_uuids_in_selected_device_uuid_list_raise_exception(
+            aggregator):
+        aggregator.device_uuid_list = [TEST_DEVICE_UUID_1]
+
+        with pytest.raises(Exception):
+            aggregator.execute_batch_commands(is_blocking=False)
+
+    @staticmethod
+    def test_execute_batch_commands_batched_command_published(aggregator):
+        uuid.uuid4 = MagicMock(return_value=TEST_TRANSACTION_ID)
+
+        aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+
+        batched_command_input = {"type": "BATCHED", "transaction_id": TEST_TRANSACTION_ID,
+                                 "aggregator_uuid": TEST_AGGREGATOR_UUID,
+                                 "batch_commands": TEST_BATCH_COMMAND_DICT}
+        batched_channel_input = f"external//aggregator/{TEST_AGGREGATOR_UUID}/batch_commands"
+
+        aggregator.execute_batch_commands(is_blocking=False)
+        aggregator.redis_db.publish.assert_called_with(batched_channel_input,
+                                                       json.dumps(batched_command_input))
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_execute_batch_commands_api_time_out_raise_exception(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking",
+                   side_effect=AssertionError):
+            aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+            aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+            with pytest.raises(RedisAggregatorAPIException):
+                aggregator.execute_batch_commands()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    @pytest.mark.parametrize("is_blocking_, trans_id_resp_buffer, expected_ret_val",
```

Comment:
Nit: why `is_blocking_` and not `is_blocking`? PEP 8 mentions using trailing underscores only in this case:
> single_trailing_underscore_: used by convention to avoid conflicts with Python keyword

I understand that you might have done it to avoid having the same parameter name, but this is not a convention we're currently using as a team and it clashes with the PEP 8 guidelines (as it would make me think that `is_blocking` is a reserved python keyword, while it isn't).
I would personally suggest to just use `is_blocking`.